### DISCUSSION TITLE
Minimum Bounding Box Area using Yolo labels and openCV

### DISCUSSION THREAD

**Discussion by thecastlingking** [ultralytics/yolov5#6588] [Ideas]
Hey Everybody,

The purpose of this python code is to get rid of bounding boxes that are too small
I made this after I couldn't find anything from searching and thought it would be useful for this community as well.

Happy Coding :)

1.         It loads in a list of pictures from a folder
2.         Then is resizes them 
3.         Then it associates the respective label file (yolov5) according to the current image
4.         Then it calculates the area of the bounding boxes
5.         Then it will display the image where:
                                                                      - 1. Red boxes are counted
                                                                      - 2. Purple boxes are not (too small)

6.         Then it saves the results to a text file with the respective image name

```
import cv2
#import matplotlib.pyplot as plt
import glob
import os

images_path = glob.glob(r"C:\Your\file\path\to\picture\*.jpg")
labels_path = glob.glob(os.path.join(os.getcwd(), r"C:\Your\file\path\to\textfile\", "*.txt"))
imglist = []
bbox_count = 0
count = 0
min_area = 200

#img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
for img_path in images_path:
    # Loading image
    img = cv2.imread(img_path)
    img = cv2.resize(img, None, fx=0.2, fy=0.2) #resize your image here to view on screen

    dh, dw, _ = img.shape
    fl = open(labels_path[count], 'r')
    data = fl.readlines()
    fl.close()
    count = count + 1
    
    
    bb_box_area = 0
    for dt in data:
        for i in dt:
            # Split string to float
            _, x, y, w, h = map(float, dt.split(' '))
        
            # Taken from https://github.com/pjreddie/darknet/blob/810d7f797bdb2f021dbe65d2524c2ff6b8ab5c8b/src/image.c#L283-L291
            # via https://stackoverflow.com/questions/44544471/how-to-get-the-coordinates-of-the-bounding-box-in-yolo-object-detection#comment102178409_44592380
            l = int((x - w / 2) * dw)
            r = int((x + w / 2) * dw)
            t = int((y - h / 2) * dh)
            b = int((y + h / 2) * dh)
            
            
            if l < 0:
                l = 0
            if r > dw - 1:
                r = dw - 1
            if t < 0:
                t = 0
            if b > dh - 1:
                b = dh - 1
            
            bb_box_area = w*dw*h*dh
            
            
            
        if ((bb_box_area)>min_area):
            cv2.rectangle(img, (l, t), (r, b), (0, 0, 255), 1)
            bbox_count+=1   
        else:
            cv2.rectangle(img, (l, t), (r, b), (255, 0, 255), 1)
        
#        print(bb_box_area)

    mystring = 'C:\\Your\\file\\path\\to\\picture\\' # Make sure this is double slash so the string captures the slash
    
    imglist.append(img_path.strip(mystring) +', ' +str(bbox_count))  #This line will append the count of the specific
    bbox_count=0
    
    # cv2.imshow('min_area', img)
    # cv2.waitKey(0)
with open('C:/your/destination/filepath/for/textfile/filename.txt', 'w') as f:
    for line in imglist:
        f.write(line)
        f.write('\n')
    f.close()
    #plt.imshow(img)
    #plt.show()
```

**Comment by glenn-jocher** [comment]
@thecastlingking candidate criteria are used here to filter candidate labels from the dataset at large during training:
https://github.com/ultralytics/yolov5/blob/9c513ca62942ba34306b9f5be3d0f0110b4435ec/utils/augmentations.py#L272-L277

**Reply by thecastlingking** [reply]
Hey @glenn-jocher 

Ahh thank you for pointing this out, but would you be able to show me how I can go about using the augmentations.py file?
(Particularly during inference)
Do I simply edit the file or do I pass something across in train.py? 

I've already read the (TTA) documentation, but it didn't reference anything about this.

I've also tried using the --help in train.py, but I wasn't able to figure out how it worked from there either.

If you're able to help I would really appreciate it and I'm sure it would help others wondering about this :)

**Reply by glenn-jocher** [reply]
@thecastlingking update arguments or function to whatever you want.
